tests/unit/memefficiency.tcl

```diff
@@ -604,24 +605,37 @@ run_solo {defrag} {
 
             # Create some clients so that they are distributed among different io threads.
             set clients {}
-            for {set i 0} {$i < 32} {incr i} {
+            for {set i 0} {$i < 8} {incr i} {
                 lappend clients [redis_client]
             }
 
             # Populate memory with interleaving key pattern of same size
-            set dummy "--[string repeat x 400]\nreturn "
-            set n 50000
-            for {set j 0} {$j < $n} {incr j} {
-                set val "$dummy[format "%06d" $j]"
-                set rr [lindex $clients [expr {int(rand() * 32)}]]
-                $rr set k$j $val
-                $rr set f$j $val
+            set dummy "[string repeat x 400]"
+            set n 10000
+            for {set i 0} {$i < [llength $clients]} {incr i} {
+                set rr [lindex $clients $i]
+                for {set j 0} {$j < $n} {incr j} {
+                    $rr set "k$i-$j" $dummy
+                }
+            }
+
+            # If io-threads is enable, verify that memory allocation is not from the main thread.
+            if {$io_threads != 1} {
+                # At least make sure that bin 448 is created in the main thread's arena.
+                r set k dummy
+                r del k
+
+                # We created 10000 string keys of 400 bytes each for each client, so when the memory
+                # allocation for the 448 bin in the main thread is significantly smaller than this,
+                # we can conclude that the memory allocation is not coming from it.
+                set malloc_stats [r memory malloc-stats]
+                if {[regexp {(?s)arenas\[0\]:.*?448[ ]+[\d]+[ ]+([\d]+)[ ]} $malloc_stats - result]} {
+                    assert_lessthan $result 200000
```

Comment:
> is that 200k bytes, or 200k allocations? 

it's 200k bytes.

> we have 8 clients and 10k allocations in each, so that's 80k allocations.

no, we create 10000 keys with 400bytes, so the allocation of bin 448 for each client is at least 4375k.

Reply:
I originally intended to verify that all arenas from 1 to 15 were over 4375k, but if I use 16 threads this test might fail because some threads might not assign the client.

Reply:
done with [`83495fb` (#13665)](https://github.com/redis/redis/pull/13665/commits/83495fb3e806446d3dfd2a2ac5935dd57ce84466)

Reply:
we create 10k keys per client. and it could be that one client will be assigned to a thread that shares arena with the main thread.
so each client allocates some 4mb, maybe instead of <200kb, it should be < 9mb?

Reply:
> we create 10k keys per client. and it could be that one client will be assigned to a thread that shares arena with the main thread. so each client allocates some 4mb, maybe instead of <200kb, it should be < 9mb?

now that the main thread does not handle client reads and writes at all, except for those clients that need to be handled in main thread, the 8 clients in this test will be evenly assigned to these io threads.

Reply:
i meant that an IO thread can share an arena with the main thread.
but i see that currently, but default the percpu_arena is disabled, and we have a static arena count of 48.
so i guess we're ok (for now)